### PR TITLE
gui: Tabify edit folder modal

### PR DESCRIPTION
**Pull request by imsodin**
Copied over from settings. Moves ignore patterns from its own modal into a tab
of the folder edit modal. Advanced settings are in a tab instead of a expandable
section and versioning gets it's own tab.
Also added modal hidden event handler to remove the anchor in the url.

### Testing

Manual poking around and changing ignore patterns.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
code wise looks ok to me, no screenshots to look at.

**Comment by imsodin** [comment]
Ups, I made them but then forgot to post :P 

All except the seconds screenshots use "Versioning" in the tabs, I changed that to "File Versioning" for consistency and clarity, but didn't redo all the screenshots.

screenshot_2018-03-17_13-47-08
screenshot_2018-03-17_15-45-09
screenshot_2018-03-17_13-47-46
screenshot_2018-03-17_13-47-57

**Comment by calmh** [comment]
Looks great

File: gui/default/syncthing/core/syncthingController.js [hunk 1]
```diff
@@ -1560,6 +1580,17 @@ angular.module('syncthing.core')
             }
             $scope.currentFolder.externalCommand = $scope.currentFolder.externalCommand || "";
 
+            $('#folder-ignores textarea').val("");
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Perhaps val("Loading...")

File: gui/default/syncthing/core/syncthingController.js [hunk 2]
```diff
@@ -1654,21 +1688,30 @@ angular.module('syncthing.core')
...
             $scope.saveConfig(function () {
-                if (!$scope.editingExisting && ignores) {
-                    $scope.saveIgnores(function () {
+                if (!$scope.editingExisting && ignores.length) {
+                    saveIgnores(ignores, function () {
                         $scope.setFolderPause(folderCfg.id, false);
                     });
                 }
             });
+
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Not needed

**Comment by imsodin** [comment]
To get status info into it ("Loading...") as requested I "added" the `$translate` service to `syncthingController.js` - seems to work (except that `go run build.go translate` does not parse it, but well). Also I don't see any reason to allow editing ignore patterns if the patterns could not be loaded, so the input field now stays disabled in that case. And paused folders don't result in an error anymore - I missed a [relevant change](https://github.com/syncthing/syncthing/pull/4822/files#diff-1df9805b2381b6288b220226f82d5dc9R1432) when cherry-picking this PR.

**Comment by calmh** [comment]
Guys. Commit subjects.

**Pull request merged**